metaflow/plugins/metadata/az_auth.py

```diff
@@ -0,0 +1,53 @@
+import atexit
+import os
+
+AUTH_PROFILE_ENV_VAR = "METAFLOW_AUTH_PROFILE"
+OBO_TOKEN_ENV_VAR = "METAFLOW_OBO_TOKEN"
+
+
+class AZAuth:
+    _fetcher = None
+    _cache = None
+
```

Comment:
Should `_fetcher` and `__cache` be a **instances** attributes instead of **class** attributed since your modifying them below in [line_36](https://github.com/fortum-tech/metaflow/pull/11/files#diff-b65dd2a5c21f5c477ed1336b6e692d9255070ed0b0e9e8fb2e5ecdd1427ee955R36) and [line_27](https://github.com/fortum-tech/metaflow/pull/11/files#diff-b65dd2a5c21f5c477ed1336b6e692d9255070ed0b0e9e8fb2e5ecdd1427ee955R27)?

Reply:
CC @fortum-vaanavil

Reply:
Ah ok, may be a small comment may help in this case ..